src/mainProgram.py

```diff
@@ -3,26 +3,34 @@
 from WekaBayesNetwork import WekaBayesNetwork
 from constants import ALL_ALGORITHMS
 import weka.core.jvm as jvm
+import traceback
+
 
 def main():
     data_provider = DataProvider(test_size=0.2)
     jvm.start()
     trainig_data = data_provider.get_weka_training_data() # for weka
-    weka_bayes_network = WekaBayesNetwork(trainig_data['labels'], ALL_ALGORITHMS['k2'])
-    weka_bayes_network.train_bayes(trainig_data['train_set'])
-    weka_bayes_network.draw_graph()
-    results = weka_bayes_network.predict_and_compare(trainig_data['test_set'])
+    try:
```

Comment:
i don't know if you should handle exception. Firstly user should be inform that he use incorrect algorithm or that he execute functions in wrong order

Reply:
Slawek: 'Firstly user should be inform that he use incorrect algorithm or that he execute functions in wrong order'

Answer: I don't understand, but program works exactly the same as you wrote.

Slawek:  'Secondly, the effect is the same as if you didn't handle, because you display stack trace in except.'

Answer: Maybe you're right, but in my opinion this is more professional approach to the situation and additionally it is related with stopping the jvm (please look at the next comment). Can you give a guarantee that crashing the program without stopping jvm by calling jvm.stop() won't create any problems in system in the future?